extract/extract.py

```diff
@@ -1209,9 +1211,26 @@ def extract_script(self, request: ServiceRequest, local: str, encoding: str):
             if len(body) <= 2:  # We can treat 2 character scripts as empty
                 continue
 
-            # Save the script and attach it as extracted
-            encoded_script = body.encode()
-            with tempfile.NamedTemporaryFile(dir=self.working_directory, delete=False) as out:
-                out.write(encoded_script)
-            extracted.append([out.name, hashlib.sha256(encoded_script).hexdigest(), encoding])
+            if script.get("language").lower() == "jscript.encode":
+                try:
+                    # The encoded VB technique can be used to encode javascript
+                    evbe_present = re.search(EVBE_REGEX, body)
+                    evbe_res = self.decode_vbe(evbe_present.groups()[0])
+                    if evbe_res and evbe_present != body:
+                        encoded_evbe_res = evbe_res.encode()
+                        with tempfile.NamedTemporaryFile(dir=self.working_directory, delete=False) as out:
+                            out.write(encoded_evbe_res)
+                        extracted.append([out.name, hashlib.sha256(encoded_evbe_res).hexdigest(), encoding])
+                except Exception:
```

Comment:
Can we log the exception that takes place pleeeeeeease

Reply:
This code was taken from extract_vbe(), so I don't know what could go wrong. We'll see.